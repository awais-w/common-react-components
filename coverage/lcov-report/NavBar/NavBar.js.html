<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for NavBar/NavBar.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">NavBar</a> NavBar.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
const NavBar = (props) =&gt; (
  &lt;div&gt;
    {React.Children.map(props.children, (child, index) =&gt;
      React.cloneElement(child, {
        isActive: props.activeStepIndex === index,
      }),
    )}
  &lt;/div&gt;
);
&nbsp;
NavBar.propTypes = {
  /**
   * Used to tell the child items whether their content is currently being shown
   */
  activeStepIndex: PropTypes.number.isRequired,
  /**
   * Collection of &lt;NavItem/&gt;'s
   */
  children: PropTypes.node.isRequired,
};
&nbsp;
export default NavBar;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 15:57:09 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
