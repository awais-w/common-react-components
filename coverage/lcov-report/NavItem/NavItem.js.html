<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for NavItem/NavItem.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">NavItem</a> NavItem.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { CSSTransition } from 'react-transition-group';
import PropTypes from 'prop-types';
import './NavItem.scss';
&nbsp;
const NavItem = ({ title, subTitle, summary, isActive, onClick, isComplete, currIndex }) =&gt; (
  &lt;div
    onClick={() =&gt; onClick({ desiredIndex: currIndex })}
    className={`step-title-container ${isActive ? 'active-item' : ''}`}&gt;
    &lt;div className='step-title-container__title'&gt;
      {isComplete ? (
        &lt;CSSTransition in timeout={300} classNames='step-title-container__complete' unmountOnExit&gt;
          &lt;div className='step-title-container__complete'&gt;â˜‘&lt;/div&gt;
        &lt;/CSSTransition&gt;
      ) : (
        ''
      )}
      {title}
    &lt;/div&gt;
&nbsp;
    {/* Only show description of step for active one */}
    {isActive &amp;&amp; &lt;div className='step-title-container__subtitle'&gt;{subTitle}&lt;/div&gt;}
&nbsp;
    {/* Show summaries on previous steps if we've got one */}
    {isComplete &amp;&amp; &lt;div className='step-title-container__summary'&gt;{summary}&lt;/div&gt;}
  &lt;/div&gt;
);
&nbsp;
NavItem.propTypes = {
  /**
   * Primary title for this step
   */
  title: PropTypes.string.isRequired,
  /**
   * Called when we click anywhere in nav item
   */
  onClick: PropTypes.func,
  /**
   * Description about whats required from the step
   */
  subTitle: PropTypes.string,
  /**
   * Show whats been completed in a previous step
   */
  summary: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Whether we're currently displaying the content for this step
   */
  isActive: PropTypes.bool.isRequired,
  /**
   * Has this stage already been completed
   */
  isComplete: PropTypes.bool,
  /**
   * The current index we've reached upto
   */
  currIndex: PropTypes.number.isRequired,
};
&nbsp;
NavItem.defaultProps = {
  onClick: null,
  summary: '',
  subTitle: '',
  isComplete: true,
};
&nbsp;
export default NavItem;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 15:57:09 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
