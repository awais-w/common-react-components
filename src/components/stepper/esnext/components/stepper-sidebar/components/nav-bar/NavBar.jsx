import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const NavBar = (props) => (
  <Fragment>
    {React.Children.map(props.children, (child, index) =>
      React.cloneElement(child, {
        isActive: props.activeStepIndex === index,
      }),
    )}
  </Fragment>
);

NavBar.propTypes = {
  /**
   * Used to tell the child items whether their content is currently being shown
   */
  activeStepIndex: PropTypes.number.isRequired,
  /**
   * Collection of <NavItem/>'s
   */
  children: PropTypes.node.isRequired,
};

export default NavBar;
