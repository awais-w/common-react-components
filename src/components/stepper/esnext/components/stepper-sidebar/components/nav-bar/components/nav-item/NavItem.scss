@import '@cmc/common-styles/src/scss/_variables.scss';

.cmc-nav-item {
  @include font-standard;
  border-bottom: 1px solid $color-grey-mid;
  padding: $double;
  cursor: pointer;

  &.locked {
    cursor: not-allowed;
  }

  &--active {
    background-color: $color-grey-light;
  }
  &__title {
    @include font-bold;
    text-transform: uppercase;
    font-size: $font-size-large;
    position: relative;
    display: flex;
    align-items: center;
    padding-right: $super-double;
  }
  &__subtitle {
    padding-top: $base;
    font-size: $font-size-medium;
  }
  &__summary {
    padding-top: $base;
  }
  &__complete {
    color: $color-green;
    transform: scale(1.25);
    position: absolute;
    right: 0;
    top: 0;
  }

  // Animation of "complete" icon
  &__complete-enter {
    opacity: 0.01;
    transform: translateY(-100%) scale(0.75);
  }
  &__complete-enter-active {
    opacity: 1;
    transform: translateY(0%) scale(1.25);
    transition: all 300ms ease-out;
  }
  &__complete-exit {
    opacity: 1;
    transform: scale(1.25);
  }
  &__complete-exit-active {
    opacity: 0;
    transform: scale(4);
    transition: all 300ms ease-in;
  }

  .cmc-icon,
  .cmc-icon-custom {
    margin-right: $half;
    transition: color 0.5s;
  }
}

.cmc-stepper--nav-minimize {
  .cmc-nav-item {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    padding: $base 0;
    &__title {
      padding-right: 0;
    }
    .cmc-icon {
      margin: 0;
      font-size: $font-size-h1;
      cursor: pointer;
    }
    .cmc-icon-custom {
      margin: 0;
      cursor: pointer;
      &::before {
        font-size: $font-size-h1;
      }
    }
    &.locked {
      cursor: pointer;
    }
    &__complete {
      & + .cmc-icon,
      & + .cmc-icon-custom {
        color: $color-green;
      }
    }
    &__summary {
      @include font-condensed-extra-bold;
      padding-top: $half;
      width: 100%;
      text-align: center;
      font-size: $font-size-small;
      line-height: 1;
    }
  }
}
