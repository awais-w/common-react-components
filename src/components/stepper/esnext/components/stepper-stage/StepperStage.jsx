/**
 * Our <Stepper/> component handles drawing content / headers
 * This just provides a cleaner interface to construct <Stepper/>
 */
import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import Waypoint from 'react-waypoint';
import { Element } from 'react-scroll';
import uuidv4 from 'uuid/v4';
import bem from '@argos/utils/esnext/bem';
import classNames from 'classnames';
import scrollTo from '../../helpers/StepperFunctions';
import './StepperStage.scss';

const cn = bem({ prefix: 'cmc-', block: 'stepper-stage' });

class StepperStage extends PureComponent {
  uniqueId;

  constructor(props) {
    super(props);
    this.uniqueId = uuidv4();
  }

  scrollToMe() {
    scrollTo({
      componentName: this.uniqueId,
      duration: this.props.allowNavigation ? 500 : 0,
      parentID: this.props.parentScrollContainer,
    });
  }

  componentDidMount() {
    this.scrollToMe();
  }

  componentDidUpdate(prevProps) {
    if (
      prevProps.activeStepIndex !== this.props.activeStepIndex &&
      this.props.activeStepIndex === this.props.stepIndex
    ) {
      this.scrollToMe();
    }
  }

  render() {
    const {
      // Provided as part of <StepperStage/> props
      title,
      subTitle,
      content,
      // Added by <Stepper />
      stageType,
      stepIndex,
      onWaypointEnter,
      shouldNavBarUpdateOnScroll,
      shouldShowStageTitle,
      allowNavigation,
      onClickPreviousStep,
    } = this.props;

    const classes = classNames(cn(), {
      disabled: stageType !== 'current',
      locked: !allowNavigation,
    });

    return (
      <Fragment>
        <div
          key={`step-${stepIndex}`}
          onClick={
            allowNavigation && stageType === 'previous' ? () => onClickPreviousStep({ desiredIndex: stepIndex }) : null
          }
          className={classes}
          name={this.uniqueId}>
          <Element key={`step-${stepIndex}`} name={`step-${stepIndex}`}>
            {/* TODO - Provide prop */}
            {shouldShowStageTitle && (
              <div className={cn('headings')}>
                <h1>{title}</h1>
                {subTitle && <h4> - {subTitle}</h4>}
              </div>
            )}
            <div className={cn('content')}>{content}</div>
          </Element>
        </div>

        {shouldNavBarUpdateOnScroll && (
          <Waypoint
            onEnter={() => {
              onWaypointEnter({ stepIndex });
            }}
          />
        )}
      </Fragment>
    );
  }
}

StepperStage.propTypes = {
  /**
   * Primary title for this step
   */
  title: PropTypes.string.isRequired,
  /**
   * Show whats been completed in a previous step
   */
  subTitle: PropTypes.string,
  /**
   * Content for the current stage being rendered
   */
  content: PropTypes.node.isRequired,
  /**
   * Set the current step ( Zero based index )
   */
  activeStepIndex: PropTypes.number,
  /**
   * Where abouts this step is located in relation to current ( So we can for example )
   * Provide the ability to navigate back to stages marked as "Previous"
   */
  stageType: PropTypes.oneOf(['current', 'previous', 'later']),
  /**
   * The index of this stage in relation to other <StepperStage />'s
   */
  stepIndex: PropTypes.number,
  /**
   * When we scroll this stage into view, this parent function will be called to update NavBar
   */
  onWaypointEnter: PropTypes.func,
  /**
   * Used to update our parent with a new activeStepID when we click on previous stages
   */
  onClickPreviousStep: PropTypes.func,
  /**
   * Helper function in <Stepper/> to scroll to a component given it's container ID
   */
  shouldNavBarUpdateOnScroll: PropTypes.bool,
  /**
   * Should we show a title above this current stage content
   */
  shouldShowStageTitle: PropTypes.bool,
  /**
   * The parent container within which the scrolling needs to happen.
   */
  parentScrollContainer: PropTypes.string,
  /**
   * Flag to indicate whether navigation is allowed from stepper stage or not.
   */
  allowNavigation: PropTypes.bool,
};

StepperStage.defaultProps = {
  subTitle: '',
  // provide default values for those provided by <Stepper/>
  activeStepIndex: 0,
  stepIndex: 0,
  stageType: 'current',
  onWaypointEnter: () => {},
  onClickPreviousStep: () => {},
  scrollTo: () => {},
  shouldNavBarUpdateOnScroll: true,
  shouldShowStageTitle: false,
  parentScrollContainer: '',
};

export default StepperStage;
