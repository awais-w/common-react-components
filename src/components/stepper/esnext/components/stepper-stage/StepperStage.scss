@import '@cmc/common-styles/src/scss/_variables.scss';

.cmc-stepper-stage {
  @include font-standard;
  min-height: 100%;
  padding: $double;
  &__headings {
    display: inline-block;

    h1,
    h4 {
      display: inherit;
    }
  }
  &__page-divider {
    margin-bottom: $large;
  }
}
