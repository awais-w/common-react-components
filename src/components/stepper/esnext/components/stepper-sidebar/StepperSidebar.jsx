import React from 'react';
import PropTypes from 'prop-types';
import bem from '@argos/utils/esnext/bem';
import { TransitionGroup } from 'react-transition-group';

// TODO - In the near future we should be importing these directly from node_modules
import IconLabelValue from '@cmc/icon-label-value/esnext';
import NavItem from './components/nav-bar/components/nav-item';
import NavBar from './components/nav-bar';

const cn = bem({ prefix: 'cmc-', block: 'stepper-container' });

const StepperSidebar = ({
  children,
  activeStepIndex,
  currentlyViewingStepIndex,
  onClickPreviousStep,
  onNavToggle,
  onOrientationToggle,
  minimizeNav,
  summaryStyle,
  allowNavigation,
  orientation,
  lastItemDoneOnReach,
}) => (
  <TransitionGroup className={cn('nav-container', null)}>
    <a className={cn('nav-toggle')} onClick={() => onNavToggle(!minimizeNav)}>
      {minimizeNav ? (
        <i className='material-icons'>swap_horiz</i>
      ) : (
        <React.Fragment>
          Minimize this column <i className='material-icons'>compare_arrows</i>
        </React.Fragment>
      )}
    </a>
    <span
      className={cn(`nav-orientation-toggle${minimizeNav ? '-bottom' : ''}`)}
      onClick={() => onOrientationToggle(orientation === 'left-side' ? 'right-side' : 'left-side')}>
      <IconLabelValue
        icon={`icon-to-${orientation === 'left-side' ? 'right' : 'left'}`}
        isCustom
        isInline
        color={minimizeNav ? 'blue' : 'white'}
        tooltipValue={`Move this column to ${orientation === 'left-side' ? 'right' : 'left'}`}
      />
    </span>
    <NavBar activeStepIndex={currentlyViewingStepIndex}>
      {React.Children.map(children, (child, currIndex) => {
        const { icon, isCustomIcon, title, subTitle, summary, shortSummary } = child.props;
        return (
          <NavItem
            key={`navitem-${title}`}
            isComplete={currIndex < activeStepIndex || (lastItemDoneOnReach && children.length === activeStepIndex + 1)}
            isActive={currIndex === activeStepIndex}
            icon={icon}
            isCustomIcon={isCustomIcon}
            title={title}
            subTitle={subTitle || null}
            summary={summary}
            shortSummary={shortSummary}
            summaryStyle={summaryStyle}
            currIndex={currIndex}
            allowNavigation={allowNavigation}
            minimizeNav={minimizeNav}
            handleClick={(e) => {
              if (minimizeNav) onNavToggle(false);
              if (allowNavigation && currIndex < activeStepIndex && !minimizeNav) {
                onClickPreviousStep(e);
              }
            }}
          />
        );
      })}
    </NavBar>
  </TransitionGroup>
);

StepperSidebar.propTypes = {
  /**
   * Set the current step ( Zero based index )
   */
  activeStepIndex: PropTypes.number.isRequired,
  /**
   * What stage we're curent viewing ( Zero based index ) - Allows our nav bar to show which content
   * we're currently viewing ( E.g when scrolling to a previous stage )
   */
  currentlyViewingStepIndex: PropTypes.number.isRequired,
  /**
   * One or more `<StepperStage/>` components.
   */
  children: PropTypes.node.isRequired,
  /**
   * Is Nav minimized
   */
  minimizeNav: PropTypes.bool,
  /**
   * Should the last step be marked as done when reached? (useful in case the last item step is confirmation)
   */
  lastItemDoneOnReach: PropTypes.bool,

  /**
   * Provide the ability to update activeStepIndex in the parent function to a certain index
   */
  onClickPreviousStep: PropTypes.func.isRequired,
  /**
   * Provide the ability to update minimizeNav in the parent function
   */
  onNavToggle: PropTypes.func.isRequired,
  /**
   * Provide the ability to update onOrientationToggle in the parent function
   */
  onOrientationToggle: PropTypes.func.isRequired,
  /**
   * When the summary should be rendered
   */
  summaryStyle: PropTypes.string,
  /**
   * Flag to indicate whether navigation is allowed from stepper stage or not.
   */
  allowNavigation: PropTypes.bool,
};

StepperSidebar.defaultProps = {
  lastItemDoneOnReach: false,
};

export default StepperSidebar;
