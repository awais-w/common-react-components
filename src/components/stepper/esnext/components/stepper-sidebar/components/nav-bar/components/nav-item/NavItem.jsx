import React from 'react';
import { CSSTransition } from 'react-transition-group';
import PropTypes from 'prop-types';
import Icon from '@cmc/icon/esnext/Icon';
import bem from '@argos/utils/esnext/bem';
import classNames from 'classnames';
import './NavItem.scss';
import NavItemConstants from './constants';

const cn = bem({ prefix: 'cmc-', block: 'nav-item' });

const NavItem = ({
  icon,
  isCustomIcon,
  title,
  subTitle,
  summary,
  shortSummary,
  isActive,
  handleClick,
  isComplete,
  currIndex,
  summaryStyle,
  allowNavigation,
  minimizeNav,
}) => {
  const classes = classNames(cn(), {
    [cn(null, 'active')]: isActive,
    locked: !allowNavigation,
  });

  return (
    <div onClick={() => handleClick({ desiredIndex: currIndex })} className={classes}>
      <div className={cn('title')}>
        {isComplete ? (
          <CSSTransition in timeout={300} classNames={cn('complete')} unmountOnExit>
            <div className={cn('complete')}>{!minimizeNav && <Icon icon={'check_circle'} />}</div>
          </CSSTransition>
        ) : (
          ''
        )}
        {icon && <Icon icon={icon} isCustom={isCustomIcon} />}
        {!minimizeNav && title}
      </div>

      {/* Only show description of step for active one */}
      {isActive && !minimizeNav && <div className={cn('subtitle')}>{subTitle}</div>}

      {/* Show summaries on previous steps if we've got one */}
      {summary &&
        ((isComplete && summaryStyle === 'after-completion') || (isActive && summaryStyle === 'real-time')) && (
          <div className={cn('summary')}>{minimizeNav ? shortSummary : summary}</div>
        )}
    </div>
  );
};
NavItem.propTypes = {
  /**
   * Icon for this step
   */
  icon: PropTypes.string.isRequired,
  /**
   * Primary title for this step
   */
  title: PropTypes.string.isRequired,
  /**
   * Called when we click anywhere in nav item
   */
  onClick: PropTypes.func,
  /**
   * Description about whats required from the step
   */
  subTitle: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Show whats been completed in a previous step
   */
  summary: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Show summary in minimzed state
   */
  shortSummary: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Whether we're currently displaying the content for this step
   */
  isActive: PropTypes.bool.isRequired,
  /**
   * Has this stage already been completed
   */
  isComplete: PropTypes.bool,
  /**
   * Is Nav minimized
   */
  minimizeNav: PropTypes.bool,
  /**
   * The current index we've reached upto
   */
  currIndex: PropTypes.number.isRequired,
  /**
   * It will display the sumarry of the step as follow:
   * - real-time: always visible
   * - after-completion: only when the step is completed
   * - hide: never visible
   */
  summaryStyle: PropTypes.oneOf(NavItemConstants.summaryStyles),
  /**
   * Determines when the stepper should allow to move to a previous step.
   */
  allowNavigation: PropTypes.bool,
};

NavItem.defaultProps = {
  icon: '',
  summary: '',
  shortSummary: '',
  subTitle: '',
  isComplete: true,
  summaryStyle: 'after-completion',
};

export default NavItem;
