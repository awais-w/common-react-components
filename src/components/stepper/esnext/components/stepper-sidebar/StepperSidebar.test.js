import React from 'react';
import { shallow } from 'enzyme';
import StepperSidebar from './StepperSidebar';
import StepperStage from '../stepper-stage';
import NavBar from './components/nav-bar';
import NavItem from './components/nav-bar/components/nav-item';

const defaultProps = {
  activeStepIndex: 0,
  currentlyViewingStepIndex: 0,
  onClickPreviousStep: jest.fn(),
  onNavToggle: jest.fn(),
  onOrientationToggle: jest.fn(),
  allowNavigation: true,
  minimizeNav: false,
  lastItemDoneOnReach: false,
};

const setup = ({ props }) => {
  const component = shallow(
    <StepperSidebar {...props}>
      <StepperStage subTitle='Stage subtitle' />
      <StepperStage />
      <StepperStage />
    </StepperSidebar>,
  );
  return {
    wrapper: component,
    getStage: (index) =>
      component
        .find(NavBar)
        .find(NavItem)
        .at(index),
  };
};

describe('StepperSidebar', () => {
  describe('base tests', () => {
    const component = setup({ props: defaultProps });
    it('should match default snapshot', () => {
      expect(component.wrapper).toMatchSnapshot();
    });
    it('exists', () => {
      expect(component.wrapper).toExist();
    });
    it('has a <NavBar> component', () => {
      expect(component.wrapper.find(NavBar)).toExist();
    });
    it('has a NavBar toggle', () => {
      expect(component.wrapper.find('a.cmc-stepper-container__nav-toggle')).toExist();
    });
    it('has a Orientation toggle', () => {
      expect(component.wrapper.find('.cmc-stepper-container__nav-orientation-toggle')).toExist();
      component.wrapper.setProps({ minimizeNav: true });
      expect(component.wrapper.find('.cmc-stepper-container__nav-orientation-toggle-bottom')).toExist();
    });
    it('stages have a default summaryStyle of after-completion when prop is not provided', () => {
      expect(component.getStage(0).props().summaryStyle).toBe('after-completion');
    });
    it('last stage, when reached, has isComplete set to true by default if lastItemDoneOnReach is true', () => {
      component.wrapper.setProps({ activeStepIndex: 2, lastItemDoneOnReach: true });
      expect(component.getStage(2).props().isComplete).toBe(true);
    });
  });

  describe('subtitle', () => {
    const component = setup({ props: defaultProps });
    it('Is null if not defined', () => {
      expect(component.getStage(1).props().subTitle).toBeNull();
    });
  });

  describe('can be clicked', () => {
    const component = setup({ props: defaultProps });
    component.wrapper.setProps({ activeStepIndex: 1 });
    it('cannot click future steps', () => {
      component
        .getStage(2)
        .props()
        .handleClick();
      expect(defaultProps.onClickPreviousStep).not.toBeCalled();
    });
    it('can click previous steps', () => {
      component
        .getStage(0)
        .props()
        .handleClick();
      expect(defaultProps.onClickPreviousStep).toBeCalled();
    });
  });

  describe('prop : onClick', () => {
    const component = setup({ props: defaultProps });
    it('Calls onNavToggle when clicked', () => {
      component.wrapper.find('.cmc-stepper-container__nav-toggle').simulate('click');
      expect(defaultProps.onNavToggle).toBeCalled();
    });
    it('Calls onOrientationToggle when clicked', () => {
      component.wrapper.find('.cmc-stepper-container__nav-orientation-toggle').simulate('click');
      expect(defaultProps.onOrientationToggle).toBeCalled();
    });
  });
});
