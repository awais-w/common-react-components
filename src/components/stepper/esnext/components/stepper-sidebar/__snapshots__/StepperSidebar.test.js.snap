// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperSidebar base tests should match default snapshot 1`] = `
<TransitionGroup
  childFactory={[Function]}
  className="cmc-stepper-container__nav-container"
  component="div"
>
  <a
    className="cmc-stepper-container__nav-toggle"
    onClick={[Function]}
  >
    Minimize this column 
    <i
      className="material-icons"
    >
      compare_arrows
    </i>
  </a>
  <span
    className="cmc-stepper-container__nav-orientation-toggle"
    onClick={[Function]}
  >
    <IconLabelValue
      color="white"
      icon="icon-to-left"
      isCustom={true}
      isInline={true}
      tooltipValue="Move this column to left"
    />
  </span>
  <NavBar
    activeStepIndex={0}
  >
    <NavItem
      allowNavigation={true}
      currIndex={0}
      handleClick={[Function]}
      icon=""
      isActive={true}
      isComplete={false}
      key="navitem-undefined/.0"
      minimizeNav={false}
      shortSummary=""
      subTitle="Stage subtitle"
      summary=""
      summaryStyle="after-completion"
    />
    <NavItem
      allowNavigation={true}
      currIndex={1}
      handleClick={[Function]}
      icon=""
      isActive={false}
      isComplete={false}
      key="navitem-undefined/.1"
      minimizeNav={false}
      shortSummary=""
      subTitle={null}
      summary=""
      summaryStyle="after-completion"
    />
    <NavItem
      allowNavigation={true}
      currIndex={2}
      handleClick={[Function]}
      icon=""
      isActive={false}
      isComplete={false}
      key="navitem-undefined/.2"
      minimizeNav={false}
      shortSummary=""
      subTitle={null}
      summary=""
      summaryStyle="after-completion"
    />
  </NavBar>
</TransitionGroup>
`;
